src/app/static/src/js/components/reports/runReportInline.vue

```diff
@@ -3,141 +3,18 @@
-            runningStatus: function() {
-                if (this.runHasCompleted) {
-                    this.stopPolling();
-                }
-            }
-        },
-        methods: {
-            confirmRun: function () {
-                this.showModal = true;
-            },
-            cancelRun: function () {
-                this.showModal = false;
-            },
-            run: function () {
-                this.showModal = false;
-                api.post(`/report/${this.report.name}/actions/run/`)
-                    .then(({data}) => {
-                        this.runningKey = data.data.key;
-                        this.runningStatus = "Run started";
-
-                        this.startPolling();
-                    })
-                    .catch(({response}) => {
-                        this.dismissRunStatus();
-                        this.runningStatus = "Error when running report";
-                        console.log(response.data);
-                    })
-                    .finally(() => {
-                        this.updateSessionStorage();
-                    });
-            },
-            startPolling: function () {
-                if (this.pollingTimer) {
-                    this.stopPolling();
-                }
-
-                this.pollingTimer = setInterval(() => {
-                        api.get(`/report/${this.report.name}/actions/status/${this.runningKey}/`)
-                            .then(({data}) => {
-                                this.runningStatus = data.data.status;
-                                this.newVersionFromRun = data.data.version;
-                            })
-                            .catch(({response}) => {
-                                this.stopPolling();
-                                this.runningStatus = "Error when fetching report status";
-                                console.log(response.data)
-                            })
-                            .finally(() => {
-                                this.updateSessionStorage();
-                            });
-                    },
-                    1500);
-            },
-            stopPolling: function () {
-                if (this.pollingTimer) {
-                    clearInterval(this.pollingTimer);
-                    this.pollingTimer = null;
-                }
-            },
-            dismissRunStatus: function () {
-                this.stopPolling();
-                Object.assign(this, initialState);
-                this.updateSessionStorage();
-            },
-            updateSessionStorage: function () {
-                if (this.runningStatus) {
-                    session.setRunningReportStatus(this.report.name, this);
-                } else {
-                    session.removeRunningReportStatus(this.report.name);
-                }
+            runReportHref: function () {
+                return `/run-report?report-name=${this.report.name}`
```

Comment:
I wonder if a much simpler approach to this in the absence of global state would simply be to store the report name (and potentially git branch/commit?) in the browser `session`, similar to the way the old version of this file used it to store the report name and status. Then we can redirect to `/run-report` and the component can just get the initial report name from the session if it exists. If this worked it would require no server-side changes at all. Any thoughts @EmmaLRussell?

Reply:
Yes, could do that. I guess I didn't suggest doing that as it's a very ephemeral property that needs to be unset as soon as the page loads as we don't want it to be used more than once. But yeah, it would simplify the backend. If we do start using session storage more, we should be structured in how we do it, e.g. keep all the props in a module, rather than just a bucket that components directly chuck stuff into!